lib/util/builder.ts

```diff
@@ -0,0 +1,48 @@
+import * as BalenaSdk from '../../typings/balena-sdk';
+import { InjectedDependenciesParam, InjectedOptionsParam } from '../balena';
+
+interface BuilderBuildFromUrlSuccessResponse {
+	started: true;
+	releaseId: number;
+}
+
+type BuilderBuildFromUrlResponse =
+	| {
+			started: false;
+			error: string;
+			message: string;
+	  }
+	| BuilderBuildFromUrlSuccessResponse;
+
+export class BuilderHelper {
+	constructor(
+		private deps: InjectedDependenciesParam,
+		private opts: InjectedOptionsParam,
+	) {}
+
+	public buildFromUrl(
+		owner: string,
+		appName: string,
+		urlDeployOptions: BalenaSdk.BuilderUrlDeployOptions,
+	) {
+		return this.deps.request
+			.send<BuilderBuildFromUrlResponse>({
+				method: 'POST',
+				url: `/v3/buildFromUrl?headless=true&owner=${owner}&app=${appName}`,
+				baseUrl: this.opts.builderUrl,
+				body: Object.assign(
+					{
+						shouldFlatten: true,
```

Comment:
Tarballs from github, once unpackaged, give you a single folder, with the contents inside that folder. If a user passes a tarball that directly wraps the content, this should be false. I would suggest adding this to the `urlDeployOptions` jsdoc signature, and just set it to `true` by default.